compiler/backend/src/org/jetbrains/kotlin/codegen/inline/inlineCodegenUtils.kt

```diff
@@ -536,10 +536,24 @@ fun isFakeLocalVariableForInline(name: String): Boolean {
 internal fun isThis0(name: String): Boolean = AsmUtil.CAPTURED_THIS_FIELD == name
 
 class InlineOnlySmapSkipper(codegen: BaseExpressionCodegen) {
-
     private val callLineNumber = codegen.lastLineNumber
 
-    fun markCallSiteLineNumber(mv: MethodVisitor) {
+    fun onInlineLambdaStart(mv: MethodVisitor, info: LambdaInfo) {
+        val firstLine = info.node.node.instructions.asSequence().mapNotNull { it as? LineNumberNode }.firstOrNull()?.line ?: -1
+        if (callLineNumber >= 0 && firstLine == callLineNumber) {
```

Comment:
What would be in case when lambda is inline function call(maybe without arguments)?
E.g.
```inlineOnlyExt { anotherInlineCall { } }```
and 
```inlineOnlyExt { inlineOnlyInlineCall {...} }```

Reply:
The first line number in those lambdas is for the inner call itself, so 65100 would be inserted.

Reply:
And if `anotherInlineCall` is not inlineOnly?

Reply:
> The first line number in those lambdas is for the inner call itself, so 65100 would be inserted.

Looks like it's should be mapped line number (am not sure of all possibilities), so 65100 shoudn't be inserted (at least for not inlineOnly inline call)

Reply:
Looks like i forgot about nop, that is generated before inline call with call site linenumber

Reply:
The nop isn't even necessary because inline functions start with fake variables which have no line number.

```
fun f() = inlineOnly { inlineOnly { } }
```
```
         0: iconst_0
         1: istore_0
         2: iconst_0
         3: istore_1
         4: iconst_0
         5: istore_2
         6: iconst_0
         7: istore_3
         8: nop
         9: nop
        10: return
      LineNumberTable:
        line 5: 0 <-- outer call site
        line 65100: 2 <-- outer lambda call
        line 5: 4 <-- inner call site
        line 65100: 6 <-- inner lambda call
        line 5: 8 <-- inside inner lambda
        ...
```

If the inner call is not `@InlineOnly`:
```
         0: iconst_0
         1: istore_0
         2: iconst_0
         3: istore_1
         4: iconst_0
         5: istore_2
         6: iconst_0
         7: istore_3
         8: nop
         9: nop
        10: nop
        11: return
      LineNumberTable:
        line 5: 0 <-- outer call
        line 65100: 2 <-- outer lambda call
        line 5: 4 <-- inner call
        line 7: 6 <-- inside inline function
        line 5: 8 <-- inside inner lambda
        ...
```

Reply:
Yes, nop is removed by optimizers further.
Please unmute
FirBlackBoxCodegenTestGenerated$CallableReference$AdaptedReferences.testInlineDefault

Reply:
That test doesn't look related to this PR, seems like it should've been unmuted in current master.

Reply:
Oh..do you mean  we can avoid it generation at all?

Reply:
Yes, but that's super minor. As you said, it's removed later anyway.